client/src/client.rs

```diff
@@ -443,44 +443,64 @@ impl Client {
         &self,
         transaction: Transaction,
     ) -> Result<HashOf<VersionedTransaction>> {
-        struct EventListenerInitialized;
-
         let client = self.clone();
-        let (event_sender, event_receiver) = mpsc::channel();
         let (init_sender, init_receiver) = mpsc::channel();
         let hash = transaction.hash();
-        let _handle = thread::spawn(move || -> eyre::Result<()> {
```

Comment:
Previously this thread was detached and values weren't being dropped.
WebSocket closing handshake is done inside event_iterator Drop impl. So connection was never closed properly

Reply:
#1899 needs to be mentioned.